### PR TITLE
AzureML: Fix ARM ID for registry environments

### PR DESCRIPTION
**Pull request by Man-MSFT**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_operation_orchestrator.py [hunk 1]
```diff
@@ -141,7 +142,7 @@ def get_asset_arm_id(
                 # CLI strips off azureml: in the schema, appending it back as required by backend
                 if (
                     azureml_type == "environments"
-                    and asset.startswith(CURATED_ENV_PREFIX)
+                    and (asset.startswith(CURATED_ENV_PREFIX) or re.match(REGISTRY_VERSION_PATTERN, f"azureml:{asset}"))
```
Comment (banibrata) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_operation_orchestrator.py:145]: How about removing this whole line# 145?

File: sdk/ml/azure-ai-ml/tests/internal_utils/unittests/test_operation_orchestrator.py [hunk 2]
```diff
@@ -186,6 +186,12 @@ def create_yaml_inline_model(tmp_path: Path, resource_group_name: str) -> Path:
 @pytest.mark.unittest
 @pytest.mark.core_sdk_test
 class TestOperationOrchestration:
+    def test_registry_environment(self, operation_orchestrator: OperationOrchestrator) -> None:
+        test_input = '//registries/mafong-registry/environments/conda_name_version_e2e/versions/1.0.2'
+        expected = 'azureml://registries/mafong-registry/environments/conda_name_version_e2e/versions/1.0.2'
+        actual = operation_orchestrator.get_asset_arm_id(test_input, azureml_type=AzureMLResourceType.Environment)
```
Comment (banibrata) [inline sdk/ml/azure-ai-ml/tests/internal_utils/unittests/test_operation_orchestrator.py:192]: "Environment" would be "ENVIRONMENT"

Comment (Man-MSFT) [reply sdk/ml/azure-ai-ml/tests/internal_utils/unittests/test_operation_orchestrator.py:192]: Oh yea good catch.

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_operation_orchestrator.py (hunk 1 above)
Comment (Man-MSFT) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_operation_orchestrator.py:145]: I'm not sure, but also hesitant to do, I don't want to accidentally miss any e2e scenarios if this was required.
Will assume that, w/e is here is needed, and only patch the bug we know about.

**Review by banibrata — APPROVED** [review]

**Review by nancy-mejia — APPROVED** [review]

**Pull request merged**